netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -258,51 +297,101 @@ public void operationComplete(ChannelFuture future) throws Exception {
         ch.closeFuture().addListener(loopReleaser);
       }
     });
-    // Bind and start to accept incoming connections.
-    ChannelFuture future = b.bind(address);
-    // We'd love to observe interruption, but if interrupted we will need to close the channel,
-    // which itself would need an await() to guarantee the port is not used when the method returns.
-    // See #6850
-    future.awaitUninterruptibly();
-    if (!future.isSuccess()) {
-      throw new IOException(String.format("Failed to bind to address %s", address), future.cause());
+    Future<Map<SocketAddress, Map<ChannelFuture,ChannelFutureListener>>> bindCallFuture =
+        bossExecutor.submit(
+            new Callable<Map<SocketAddress, Map<ChannelFuture, ChannelFutureListener>>>() {
+          @Override
+          public Map<SocketAddress, Map<ChannelFuture, ChannelFutureListener>> call() {
+            Map<SocketAddress, Map<ChannelFuture, ChannelFutureListener>> bindFutures =
+                new HashMap<>();
+            for (SocketAddress address: addresses) {
+                ChannelFuture future = b.bind(address);
+                channelGroup.add(future.channel());
+                final InternalInstrumented<SocketStats> listenSocketStats =
+                    new ListenSocket(future.channel());
+                future.addListener(new ChannelFutureListener() {
+                  @Override
+                  public void operationComplete(ChannelFuture future) throws Exception {
+                    channelz.addListenSocket(listenSocketStats);
+                    listenSocketStatsList.add(listenSocketStats);
+                  }
+                });
+                ChannelFutureListener closeListener = new ChannelFutureListener() {
+                  @Override
+                  public void operationComplete(ChannelFuture future) throws Exception {
+                    if (!future.isSuccess()) {
+                      log.log(Level.WARNING, "Error closing server channel", future.cause());
```

Comment:
Drop this. This listener isn't responsible for closing, just interested to know when it happens. The code that actually closes should be handling this error.